services/wallet/router/router_helper.go

```diff
@@ -256,9 +256,8 @@ func (r *Router) evaluateAndUpdatePathDetails(ctx context.Context, path *routes.
 	l1TxFeeWei := big.NewInt(0)
 	l1ApprovalFeeWei := big.NewInt(0)
 
-	needL1Fee := path.FromChain.ChainID != walletCommon.EthereumMainnet &&
-		path.FromChain.ChainID != walletCommon.EthereumSepolia &&
-		path.FromChain.ChainID != walletCommon.AnvilMainnet
+	needL1Fee := path.FromChain.ChainID == walletCommon.OptimismMainnet ||
+		path.FromChain.ChainID != walletCommon.OptimismSepolia
```

Comment:
```suggestion
		path.FromChain.ChainID == walletCommon.OptimismSepolia
```

Reply:
Yes, I know, you commented on it before I fixed it. :)